db/sql/0_msar.sql

```diff
@@ -662,6 +662,45 @@ SELECT EXISTS (SELECT 1 FROM pg_attribute WHERE attrelid=tab_id AND attname=col_
 $$ LANGUAGE SQL RETURNS NULL ON NULL INPUT;
 
 
+CREATE OR REPLACE FUNCTION msar.get_schemas() RETURNS jsonb AS $$/*
+Return a json array of objects describing the user-defined schemas in the database.
+
+PostgreSQL system schemas are ignored.
+
+Internal Mathesar-specifc schemas are INCLUDED. These should be filtered out by the caller. This
+behavior is to avoid tight coupling between this function and other SQL files that might need to
+define additional Mathesar-specific schemas as our codebase grows.
+
+Each returned JSON object in the array will have the form:
+  {
+    "oid": <int>
+    "name": <str>
+    "description": <str|null>
+    "table_count": <int>
+  }
+*/
+SELECT jsonb_agg(schema_data)
+FROM (
+  SELECT 
+    s.oid AS oid,
+    min(s.nspname) AS name,
+    min(d.description) AS description,
+    COALESCE(count(c.oid), 0) AS table_count
+  FROM pg_catalog.pg_namespace s
+  LEFT JOIN pg_catalog.pg_description d ON
+    d.objoid = s.oid AND
+    d.objsubid = 0
+  LEFT JOIN pg_catalog.pg_class c ON
+    c.relnamespace = s.oid AND
+    c.relkind = 'r'
```

Comment:
I think this filtering move warrants a comment about why it's in the `JOIN` rather than `WHERE` clause. It's not completely nonstandard, but it's not super common or obvious either. I spent longer than I'm proud of working through what would go wrong if it were just in the `WHERE` clause.

Reply:
Fixed in e79f31c2e1f4a36b4cb68a67bcea92d49bb917fb

> it's not super common or obvious

For what it's worth: I tend to use the `ON` clause for filtering conditions any time I'm using `LEFT JOIN`. So I stuff things in the `ON` clause quite often, as a pattern. Within the `WHERE` clause I typically only reference tables from `FROM` and `JOIN` clauses — or sometimes I'll reference the pk of a `LEFT JOIN` table when I'm doing an exclusion join.

There might be exceptions to that pattern. But I just mention this to give you a window into my thought process when writing queries. Like, if I see a non-pk column from a `LEFT JOIN` table referenced in `WHERE`, it'll raise my eyebrows and I'll look for a bug.